src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -337,6 +358,198 @@ public string GetFileShareLocationForReleaseJson(DownloadedBuild build)
+        };
+
+        const string identitiesDirectoryName = "identities";
+        const string nupkgDirectoryName = "nupkgs";
+        const string sympkgsDirectoryName = "sympkgs";
+        const string allNupkgsFileName = "nupkgs-all-just-for-reference.txt";
+        const string nupkgsFileNamePrefix = "nupkgs-";
+        const string sympkgsFileName = "sympkgs-all.txt";
+        const string packagesSubDirs = "packages";
+        const string symPackagesSubDirs = "assets/symbols";
+
+        /// <summary>
+        ///     Create the nupkg layout required for the final release.
+        /// </summary>
+        /// <param name="downloadedBuilds">List of downloaded builds</param>
+        /// <param name="outputDirectory">Output directory to create the nupkg layout</param>
+        /// <remarks>
+        ///     Generates the nupkg layout that used by the release pipeline. The layout is as follows:
+        ///     identities\
+        ///         csv files containing package name + version, named with short name of repo
+        ///     nupkgs\
+        ///         repo-shortname\
+        ///             packages\
+        ///                 shipping packages from repo
+        ///     sympkgs\
+        ///         repo-shortname\
+        ///             assets\
+        ///                 symbols\
+        ///                     symbol packages from repo
+        ///     nupkgs-all-just-for-reference.txt - file list of all packages, 
+        ///     nupkgs-aspnet.txt - file list of all packages that aspnetcore owns on nuget.org
+        ///     nupkgs-core.txt - file list of all packages that dotnet core owns on nuget.org
+        ///     sympkgs-all.txt - file list of all symbol packages
+        /// </remarks>
+        /// <returns>Async task</returns>
+        private void CreateReleasePackageLayout(List<DownloadedBuild> downloadedBuilds, string outputDirectory)
+        {
+            if (_options.DryRun || !_options.ReleaseLayout)
+            {
+                return;
+            }
+
+            // Identify the subdirectories. Do not need to create these now because
+            // we will create them as they go
+            string nupkgDirectory = Path.Combine(outputDirectory, nupkgDirectoryName);
+            string sympkgsDirectory = Path.Combine(outputDirectory, sympkgsDirectoryName);
+
+            // Universal text file content
+            StringBuilder allNupkgsFileContent = new StringBuilder();
+            StringBuilder sympkgsFileContent = new StringBuilder();
+
+            // Identity Csv contents
+            Dictionary<string, StringBuilder> identitiesFileContents = new Dictionary<string, StringBuilder>();
+
+            // core/wcf/aspnet nupkg lists
+            Dictionary<string, StringBuilder> nupkgFileContents = new Dictionary<string, StringBuilder>();
+
+            // Walk each input repo build and start filling out the directories
+            foreach (var downloadedBuild in downloadedBuilds)
```

Comment:
I think it wouldn't be worth doing this or the inner loops in parallel? Or perhaps using async in some places?

Reply:
Nah. This is just copying files from A->B locally. It's very fast today.